src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -192,19 +181,21 @@ internal async Task ConnectAsyncJavaScript(Uri uri, CancellationToken cancellati
                         if (!cancellationToken.IsCancellationRequested)
                         {
                             // Change internal _state to 'Connected' to enable the other methods
-                            if (Interlocked.CompareExchange(ref _state, (int)InternalState.Connected, (int)InternalState.Connecting) != (int)InternalState.Connecting)
+                            int prevState = _state;
+                            _state = _state == (int)InternalState.Connecting ? (int)InternalState.Connected : _state;
+                            if (prevState != (int)InternalState.Connecting)
                             {
                                 // Aborted/Disposed during connect.
-                                tcsConnect.TrySetException(new ObjectDisposedException(GetType().FullName));
+                                _tcsConnect.TrySetException(new ObjectDisposedException(GetType().FullName));
                             }
                             else
                             {
-                                tcsConnect.SetResult();
+                                _tcsConnect.SetResult();
                             }
                         }
                         else
                         {
-                            tcsConnect.SetCanceled(cancellationToken);
+                            _tcsConnect.SetCanceled(cancellationToken);
```

Comment:
Elsewhere you've used TrySet*... is it intentional for these two that they're using the non-Try variants?

Reply:
Addressed in recent commits